docs/getting-started.md

```diff
@@ -0,0 +1,461 @@
+# Contributing to OSTree: Getting Started
+
+The following guide is about OSTree forking, building, adding a command, testing the command, and submitting the change.
+
+- [Getting Started](#getting-started)
+- [Building OSTree](#building-ostree)
+    - [Install Build Dependencies](#install-build-dependencies)
+    - [OSTree Build Commands](#ostree-build-commands)
+- [Testing a Build](#testing-a-build)
+    - [Testing in a Container](#testing-in-a-container)
+    - [Testing in a Virtual Machine](#testing-in-a-virtual-machine)
+- [Tutorial: Adding a basic builtin command to OSTree](#tutorial-adding-a-basic-builtin-command-to-ostree)
+    - [Modifying OSTree](#modifying-ostree)
+    - [OSTree Tests](#ostree-tests)
+    - [Submitting a Patch](#submitting-a-patch)
+    - [Returning Workflow](#returning-workflow)
+
+---
+
+## Getting Started
+
+Fork https://github.com/ostreedev/ostree, then run the following commands.
+
+```bash
+$ git clone https://github.com/<username>/ostree
+$ git remote add upstream https://github.com/ostreedev/ostree
+$ git checkout master
+$ git branch --set-upstream-to=upstream/master master
+```
+Make a branch from master for your patch.
+
+```bash
+$ git checkout -b <name-of-branch>
+$ git branch --set-upstream-to=upstream/master <name-of-branch>
+```
+
+## Building OSTree
+
+### Install Build Dependencies
+
+Execute one of the following group commands as superuser depending on your machine's package manager.
+
+For Fedora:
+
+```bash
+dnf install @buildsys-build dnf-plugins-core && \
+dnf builddep ostree
+```
+
+For CentOS:
+
+```bash
+yum install @buildsys-build dnf-plugins-core && \
```

Comment:
s/dnf-plugins-core/yum-utils/

CentOS unfortunately doesn't have `@buildsys-build` -- see: https://github.com/ostreedev/ostree/blob/521e0ec3ac87b539190f3a6cc310237fbaf39517/ci/libpaprci/libbuild.sh#L39

Reply:
Thanks for pointing it out.

I supposed CentOS would still need `dnf-plugins-core`, since this will allow [builddep](https://centos.pkgs.org/7/puias-unsupported-x86_64/dnf-plugins-core-0.1.5-3.sdl7.noarch.rpm.html) to run. 

Let me know if I should keep `dnf-plugins-core` as one of the commands to run for CentOS, or if I should specify the individual packages as stated in libbuild.sh and remove the directions to install `dnf-plugins-core` and the `yum-builddep ostree` command.

Reply:
Right, that line would be:

```
yum install yum-utils <list of base pkgs in libbuild.sh>
```

`yum builddep` is provided by `yum-utils` so you don't need `dnf-plugins-core`.